File: PA-2/products/productdetails.php
<!doctype html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="author" content="">
  <meta name="description" content="">
  <meta name="viewport" content="width=device-width, initial-scale=1">


  <title>Product Page</title>

  <!-- external css file -->
  <script src="https://use.fontawesome.com/releases/v5.11.1/js/all.js"></script>
  <link href="../css/style.css" rel="stylesheet">
  <link href="../css/productdetails.css" rel="products stylesheet">
  <link href="https://www.w3schools.com/w3css/4/w3.css" rel="stylesheet">

  <!-- external javascript -->
  <script src="../js/productdetails.js"></script>
  <script src="https://code.jquery.com/jquery-2.1.1.min.js" type="text/javascript"></script>

  <!--internal javascript -->
  <script type="text/javascript">
    $(document).ready(function() {
      /* Go to Top Button */
      var elmClass = '.gotopbtn'; // Adjust this accordingly.

      //Check to see if the window is top if not then display button
      $(window).scroll(function() {
        if ($(this).scrollTop() > 300) { // 300px from top
          $(elmClass).fadeIn();
        } else {
          $(elmClass).fadeOut();
        }
      });

      //Click event to scroll to top
      $(elmClass).click(function() {
        $('html, body').animate({
          scrollTop: 0
        }, 800);
        return false;
      });
      /*----------------------------*/
    });

    function openSearch() {
      document.getElementById("myOverlay").style.display = "block";
    }

    function closeSearch() {
      document.getElementById("myOverlay").style.display = "none";
    }

    function expandIMG(imgs) {
      var expanded = document.getElementById("expanded");
      expanded.src = imgs.src;
      expanded.parentElement.style.display = "block";
    }

    // -----------Ajax functions--------------
    /*
    For state autofill
    */
    $(document).ready(function(){
      $("#state").keyup(function(){
        var state = $('#state').val();
        var price = $('#price').text();

        $.ajax({
        type: "POST",
        url: "getState.php",
        // data:'keyword='+$(this).val(),
        data:{"keyword": state, "productprice": price},
        beforeSend: function(){
          $("#state").css("background","#FFF url(../img/LoaderIcon.gif) no-repeat 165px");
        },
        success: function(data){
          // document.getElementById("autofill-box").innerHTML = data;
          $("#autofill-state").show();
          $("#autofill-state").html(data);
          $("#state").css("background","#FFF");
        }
        });
      });
    });

    /*
    For zipcode autofill
    */
    $(document).ready(function(){
      $("#zip").keyup(function(){
        var zip = $('#zip').val();
        var price = $('#price').text();

        $.ajax({
        type: "POST",
        url: "getZip.php",
        // data:'keyword='+$(this).val(),
        data: {"keyword":zip, "productprice":price},
        beforeSend: function(){
          $("#zip").css("background","#FFF url(../img/LoaderIcon.gif) no-repeat 165px");
        },
        success: function(data){
          // document.getElementById("autofill-zip").innerHTML = data;
          $("#autofill-zip").show();
          $("#autofill-zip").html(data);
          $("#zip").css("background","#FFF");
        }
        });
      });
    });

    $(document).ready(function() {
      $("#qty").keyup(function() {
        var quantity = $('#qty').val();
        var price = $('#price').text();
        var tax = $('#tax').text();

        $.ajax({
          type: "POST",
          url: "getQuantity.php",
          data: {"quantity": quantity, "productprice": price, "tax": tax},
          success: function(data) {
            console.log("test");
            $("#total").html(data);
          }
        });
      });
    });

    function selectState(val, productprice) {
      $("#state").val(val);
      getTax(val, productprice);
      $("#autofill-state").hide();
    }

    function selectZip(val, productprice) {
      $("#zip").val(val);
      fillZipInfo(val, productprice);
      $("#autofill-zip").hide();
    }

    function getTax(state, productprice) {
      var xmlhttp = new XMLHttpRequest();
      xmlhttp.onreadystatechange = function() {
        if (xmlhttp.readyState == 4 && xmlhttp.status == 200) {
          var info = xmlhttp.responseText.split(";");
          document.getElementById("tax").innerHTML = info[0];
          document.getElementById("total").innerHTML = info[1];
          document.getElementById("qty").value = 1;
        }
        else {
            document.getElementById("tax").innerHTML = "error";
        }
      };
      xmlhttp.open("GET", "getTax.php?state=" + state + "&productprice=" + productprice, true);
      xmlhttp.send(null);
    }

    function fillZipInfo(zip, productprice) {
      var xmlhttp = new XMLHttpRequest();
      xmlhttp.onreadystatechange = function() {
        if (xmlhttp.readyState == 4 && xmlhttp.status == 200) {
          var info = xmlhttp.responseText.split(",");
          document.getElementById("state").value = info[1];
          document.getElementById("city").value = info[2];
          getTax(info[1], productprice);
        }
      };
      xmlhttp.open("GET", "fillAll.php?zip=" + zip, true);
      xmlhttp.send(null)
    }

  </script>

  <!-- internal css -->
  <style>
  </style>

</head>

<body>

  <!-- Overlay for Search button -->
  <div id="myOverlay" class="overlay">
    <span class="closebtn" onclick="closeSearch()" title="Close Overlay">×</span>
    <div class="overlay-content">
      <form action="#">
        <input type="text" placeholder="Search.." name="search">
        <button type="submit"><i class="fa fa-search"></i></button>
      </form>
      <br>
    </div>
  </div>

  <div class="main-container">
    <!--<p>hello world!</p>-->

    <!-- Nav -->
    <!-- https://www.w3schools.com/howto/howto_css_topnav_right.asp-->
    <?php
      include('product_header.php');
    ?>

    <!-- Main -->
    <section>
      <div class="main">

        <?php
          //<p><a href="productcategory_switch.html">Back to Nintendo Switch Product Page</a></p>
          $pageFrom = $_GET['pageFrom'];
          $productCategory = $_GET['productCategory'];
          echo "<p><a href='".$pageFrom."'>Back to ".$productCategory." Product Page</a></p>";
        ?>

        <!-- Product Detail Card -->
        <?php

          include('../database.php');
          //get product given the product ID
          $productID = $_GET['productID'];
          $products = getProduct($productID);
          $product = $products[0]; //retrieve the first element of array

          //get main image
          //the first element in array is the main image
          $imageLinksArray = explode(",",$product['ImageLinks']);
          $mainImagePath = $imageLinksArray[0];

        ?>

        <div class="w3-card-4 productdetail-card" style="width:48%; float:left;">

          <?php
            //<center><img src="../img/products/zelda/zelda.jpg" alt="p1" style="width:50%; padding-top: 10px"></center>
            echo "<center><img src='".$mainImagePath."' alt='p1' style='width:50%; padding-top: 10px'></center>"
          ?>

          <div class="main">
            <div class="expandedImgContainer">
              <span onclick="this.parentElement.style.display='none'" class="expanded-closebtn">&times;</span>
              <img id="expanded" style="width: 100%">
            </div>

            <div class="row">
            <?php
              for($i = 1; $i < count($imageLinksArray); $i++){
                echo "
                      <div class='column'>
                        <img src='".$imageLinksArray[$i]."' alt='screenshot0".$i."' style='width:100%' onclick='expandIMG(this);'>
                      </div>
                    ";
              } //end of for
            ?>
            </div>
          </div>

          <?php
            echo "
                    <div class='main'>
                      <h4 id='productname'><b>".$product['ProductName']."</b></h4>
                      <p>Platform: ".$product['Platform']."</p>
                      <p>Category: ".$product['Category']."</p>
                      <p id='productprice'>Price: ".$product['Price']."</p>
                      <p>Currently In Stock: ".$product['Quantity']."</p>
                      <p>Product ID: ".$product['ProductID']."</p>
                      <p>Description: ".$product['Description']."</p>
                    </div>
                  ";
          ?>

        </div>
        <!-- end of product detail card div -->


        <!-- Order Form Card -->
        <div class="w3-card-4 orderform-card" style="width:48%; float:right;">
          <!--<form action="" name="orderform" onsubmit="return(validate());">-->
          <form method="post" action="ordervalidation.php" name="orderform" onsubmit="return(validate());">
            <div class="col-50">
              <strong><h2>Order Form</h2></strong>
              <h4>Shipping Information</h4>

              <div class="row">
                <div class="col-50">
                  <label for="cname">First Name</label>
                  <input type="text" id="fname" name="firstname" placeholder="John" required>
                </div>

                <div class="col-50">
                  <label for="cname">Last Name</label>
                  <input type="text" id="lname" name="lastname" placeholder="Doe" required>
                </div>
              </div>

              <label for="adr">Address</label>
              <input type="text" id="adr" name="address" placeholder="542 W. 15th Street" required>

              <label for="city">City</label>
              <input type="text" id="city" name="city" placeholder="New York" required>

              <div class="row">
                <div class="col-50">
                  <label for="state">State</label>
                  <input type="text" id="state" name="state" placeholder="State Name" required/>
                  <div id="autofill-state"></div>
                </div>

                <div class="col-50">
                  <label for="zip">Zip</label>
                  <input type="text" id="zip" name="zip" placeholder="10001" required>
                  <div id="autofill-zip"></div>
                </div>

              </div>

              <label for="Shippng Method">Shipping Method:</label>

              <select class="select-css" id="Shipping Method" name="shippingmethod">
                <option value="overnight">Overnight</option>
                <option value="2-day">2-Days Expedited</option>
                <option value="6-day">6-Days Ground</option>
              </select>

              <h4>Billing Information</h4>
              <div class="row">
                <div class="col-50">
                  <label for="productid">Product ID</label>
                  <?php
                    //<input type="text" id="productid" name="productid" placeholder="1GA13" pattern=".{5}" required>
                    echo"<input type='text' id='productid' name='productid' value='".$product['ProductID']."' readonly required>"
                  ?>
                </div>
                <div class="col-50">
                  <label for="qty">Quantity (5 Max)</label>
                  <input type="number" id="qty" name="qty" placeholder="1" min="0" max="5" required>
                </div>
              </div>

              <div class="row">
                <div class="col-50">
                  <p>Tax Rate: <span id="tax"> 0 </span>%</p>
                </div>
                <div class="col-50">
                  <p>Total Price: $
                    <span id="total">
                      <?php echo $product['Price'] ?>
                    </span></p>
                </div>
              </div>
              <br>

              <label for="ccnum">Credit card number</label>
              <input type="text" id="ccnum" name="cardnumber" placeholder="1111-2222-3333-4444" required>

              <label for="expmonth">Exp Month</label>
              <input type="text" id="expmonth" name="expmonth" placeholder="September" required>

              <div class="row">
                <div class="col-50">
                  <label for="expyear">Exp Year</label>
                  <!--<input type="text" id="expyear" name="expyear" placeholder="2018" pattern="[2020-2100]">-->
                  <input type="text" id="expyear" name="expyear" placeholder="2018" min="2018" max="2100" required>
                </div>

                <div class="col-50">
                  <label for="cvv">CVV</label>
                  <input type="text" id="cvv" name="cvv" placeholder="352" pattern="[0-9]{3}" required>
                </div>
              </div>

              <h4>Receipt/Tracking</h4>

              <label for="phone">Phone Number</label>
              <input type="tel" id="phone" name="phone" placeholder="1-555-666-7777" pattern="[0-9]{1,}-[0-9]{3}-[0-9]{3}-[0-9]{4}" required>

              <label for="email">Email</label>
              <input type="text" id="email" name="email" placeholder="john@example.com" required>

              <input type="submit" name="submit" value="Submit" />

            </div>
          </form>
        </div>
        <!-- end of order form card div -->

        <br>
        <p id="price", hidden=true><?php echo $product["Price"]; ?></p>

      </div>
      <!-- end of main div -->
    </section>

    <!-- Footer -->
    <?php
      include('../footer.php');
    ?>
  </div>

  <!-- go to top button -->
  <a class="gotopbtn" href="#"><span class="fas fa-caret-square-up"></span></a>

</body>

</html>

File: PA-3/src/com/gamehub/ProductDetails.java
package com.gamehub;

import java.io.IOException;
import java.io.PrintWriter;
import java.util.Map;

import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class ProductDetails extends HttpServlet {
	PrintWriter output;
	HttpServletResponse response;
	
	public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
		renderPage(req,res);
	}
	public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException {
		renderPage(req,res);
	}
	
	private void renderPage(HttpServletRequest req, HttpServletResponse res) throws IOException {
		response = res;
		response.setContentType("text/html;charset=UTF-8");
		p("");
		p("<!doctype html>");
		p("<html lang=\"en\">");
		
		//HEAD TAG
		p("<head>");
		p("<meta charset=\"utf-8\"> \n" +
		  "<meta name=\"author\" content=\"\">\n" +
		  "<meta name=\"description\" content=\"\">\n" +
		  "<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n" +


		  "<title>Product Page</title>\n" +

		  "<!-- external css file -->\n" +
		  "<script src=\"https://use.fontawesome.com/releases/v5.11.1/js/all.js\"></script>\n" +
		  "<link href=\"./css/style.css\" rel=\"stylesheet\">\n" +
		  "<link href=\"./css/productdetails.css\" rel=\"products stylesheet\">\n" +
		  "<link href=\"https://www.w3schools.com/w3css/4/w3.css\" rel=\"stylesheet\">\n" +

		  "<!--internal javascript -->\n" +
		  "<script type=\"text/javascript\">\n" +
		    "$(document).ready(function() {\n" +
		      "/* Go to Top Button */\n" +
		      "var elmClass = '.gotopbtn'; // Adjust this accordingly.\n" +

		      "//Check to see if the window is top if not then display button\n" +
		      "$(window).scroll(function() {\n" +
		        "if ($(this).scrollTop() > 300) { // 300px from top\n" +
		          "$(elmClass).fadeIn();\n" +
		        "} else {\n" +
		          "$(elmClass).fadeOut();\n" +
		        "}\n" +
		      "});\n" +

		      "//Click event to scroll to top\n" +
		      "$(elmClass).click(function() {\n" +
		        "$('html, body').animate({\n" +
		          "scrollTop: 0\n" +
		        "}, 800);\n" +
		        "return false;\n" +
		      "});\n" +
		      "/*----------------------------*/\n" +
		    "});\n" +

		    "function openSearch() {\n" +
		      "document.getElementById(\"myOverlay\").style.display = \"block\";\n" +
		    "}\n" +

		    "function closeSearch() {\n" +
		      "document.getElementById(\"myOverlay\").style.display = \"none\";\n" +
		    "}\n" +

		    "function expandIMG(imgs) {\n" +
		      "var expanded = document.getElementById(\"expanded\");\n" +
		      "expanded.src = imgs.src;\n" +
		      "expanded.parentElement.style.display = \"block\";\n" +
		    "}\n" +
		  "</script>\n" +

		  "<!-- external javascript -->\n" +
		  "<script src=\"./js/productdetails.js\"></script>\n" +

		  "<!-- internal css -->\n" +
		  "<style>\n" +
		  "</style>\n" +
		  "");
		p("</head>");

		//BODY TAG
		p("<body>");

		p("<!-- Overlay for Search button --> \n" +
		  "<div id=\"myOverlay\" class=\"overlay\">\n" +
		    "<span class=\"closebtn\" onclick=\"closeSearch()\" title=\"Close Overlay\">×</span>\n" +
		    "<div class=\"overlay-content\">\n" +
		      "<form action=\"#\">\n" +
		        "<input type=\"text\" placeholder=\"Search..\" name=\"search\">\n" +
		        "<button type=\"submit\"><i class=\"fa fa-search\"></i></button>\n" +
		      "</form>\n" +
		    "</div>\n" +
		  "</div>\n" +


		  "<div class=\"main-container\">\n" +
		    "<!--<p>hello world!</p>-->\n" +

		    "<!-- Nav -->\n" +
		    "<!-- https://www.w3schools.com/howto/howto_css_topnav_right.asp-->\n" +
		    "<header>\n" +
		      "<div class=\"topnav\">\n" +
		        "<!-- Left-aligned links -->\n" +
		        "<!--<a id=\"topnav-logo\" href=\"index.html\"><img src=\"img/logo_placeholder.png\" alt=\"Logo\"></a>-->\n" +
		        "<a href=\"./index.html\">Home</a>\n" +
		        "<a class=\"active\" href=\"./products.html\">Products</a>\n" +

		        "<!-- https://www.w3schools.com/howto/howto_css_subnav.asp -->\n" +
		        "<!--<a href=\"#about\">About</a>-->\n" +
		        "<div class=\"subnav\">\n" +
		          "<button class=\"subnavbtn\">About <i class=\"fa fa-caret-down\"></i></button>\n" +
		          "<div class=\"subnav-content\">\n" +
		            "<a href=\"./company.html\">Company</a>\n" +
		            "<a href=\"./team.html\">Team</a>\n" +
		            "<!-- <a href=\"#careers\">Careers</a> -->\n" +
		            "<a href=\"./contact.html\">Contact</a>\n" +
		          "</div>\n" +
		        "</div>\n" +

		        "<!-- Right-aligned links-->\n" +
		        "<div class=\"topnav-right\">\n" +
		          "<!--<a href=\"#search\">Search</a>-->\n" +
		          "<button class=\"searchBtn\" onClick=\"openSearch()\" style=\"float:left;\"><i class=\"fas fa-search\"></i> Search</button>\n" +
		          "<a href=\"./shoppingcart.html\"><i class=\"fas fa-shopping-cart\"></i> Shopping Cart</a>\n" +
		        "</div>\n" +

		      "</div>\n" +
		    "</header>\n" +
		    "");

			//MAIN SECTION
			
			String pageFrom = req.getParameter("pageFrom");
			String productCategory = req.getParameter("productCategory");
			String productID = req.getParameter("productID");
			
			Map<String, Object> product = Database.getProduct(productID);
			
			//Retrieve the main image. This is the first image in the array.
			String imageLinks = (String) product.get("ImageLinks"); 
			String[] imageLinksArray = imageLinks.split(",");
			String mainImage = imageLinksArray[0];
			
		    p("<!-- Main -->\n" +
		    "<section>\n" +
		      "<div class=\"main\">\n" +
		        "<p><a href=\""+pageFrom+"\">Back to "+productCategory+" Product Page</a></p>\n" +

		        "<!-- Product Detail Card -->\n" +
		        "<div class=\"w3-card-4 productdetail-card\" style=\"width:48%; float:left;\">\n" +
		          "<center><img src=\""+mainImage+"\" alt=\"p1\" style=\"width:50%; padding-top: 10px\"></center>\n" +

		          "<div class=\"main\">\n" +
		            "<div class=\"expandedImgContainer\">\n" +
		              "<span onclick=\"this.parentElement.style.display='none'\" class=\"expanded-closebtn\">&times;</span>\n" +
		              "<img id=\"expanded\" style=\"width: 100%\">\n" +
		            "</div>\n" +
		            "");  
		    
		            p("<div class=\"row\">");
		              for(int i=0; i<imageLinksArray.length; i++) {
		            	  p("<div class=\"column\">\n" +
		                    "<img src=\""+imageLinksArray[i]+"\" alt=\"screenshot0"+i+"\" style=\"width:100%\" onclick=\"expandIMG(this);\">\n" +
		                    "</div>\n");
		              }
					p("</div>\n" +
		          "</div>\n" +
		          "");
		            
		          p("<div class=\"main\">\n" +
		            "<h4 id=\"productname\"><b></b></h4>\n" +
		            "<p>Platform: "+product.get("Platform")+"</p>\n" +
		            "<p>Category: "+product.get("Category")+"</p>\n" +
		            "<p id=\"productprice\">Price: "+product.get("Price")+"</p>\n" +
		            "<p>Currently In Stock: "+product.get("Quantity")+"</p>\n" +
					"<p>Product ID: "+product.get("ProductID")+"</p>\n" +
		            "<p>Description: "+product.get("Description")+"</p>\n" +
		          "</div>\n" +
		        "</div>\n" +
		        "<!-- end of product detail card div -->\n" +

		        
		        "<br>\n" +

		      "</div>\n" +
		      "<!-- end of main div -->\n" +
		    "</section>\n" +
		    "");

		  //FOOTER TAG
		 p("<!-- Footer --> \n" +
		    "<!-- \n" +
		      "<div class=\"footer\"> \n" +
		        "<p>Footer</p> \n" +
		      "</div> --> \n" +
		    "<footer> \n" +
		      "<p>Acme Web Design, Copyright &copy; 2020</p> \n" +
		    "</footer> \n" +
		  "</div> \n" +

		  "<!-- go to top button --> \n" +
		  "<a class=\"gotopbtn\" href=\"#\"><span class=\"fas fa-caret-square-up\"></span></a> \n" +
		  "");
		p("</body>");
		p("</html>");


		//clean up
		output = null;
		response = null;

	}
	
	/**
	 * Helper Method
	 * @param message
	 * @throws IOException
	 */
	private void p(String message) throws IOException{
		if(output!=null) {
			output.println(message);
		} else {
			output = response.getWriter();
		}
	}
	
}
